packages/compass-aggregations/src/components/stage-editor/stage-editor.jsx

```diff
@@ -83,11 +80,6 @@ class StageEditor extends Component {
     if (this.props.stageOperator !== prevProps.stageOperator && this.editor) {
       // Focus the editor when the stage operator has changed.
       this.editor.focus();
-
-      // When the underlying stage operator changes, re-run the preview.
-      if (this.props.isAutoPreviewing) {
-        this.debounceRun();
-      }
```

Comment:
@Anemy when you have a moment can you take a look at this PR and here specifically give me some context on why we were running the stage during the operator change? I don't see us doing this before and run into conflicts around this part of the code when trying to merge with latest changes on main. All the tests seems to be passing now, but I am not sure if I accidentally broke something and how to take your recent changes into account for the refactoring on `onStateChange` usage in the component

Reply:
Did some manual testing and doesn't seem like any behavior is changing in Compass when this is removed

Reply:
Sounds good, previously we were re-running the stage preview when the operator changed by doing an explicit setting of the editor text which would cause the `onChange` to fire from this `componentDidMount`. 
Here's the diff: https://github.com/mongodb-js/compass/pull/3048/files#diff-4017e96319d72311f3b5911e298191068ffb3ac25ed4d7166e288d5aa5ae4b75R84
I added this `debounceRun` call because when the stage operator changes we want the preview to update if possible I think. I'll try out your changes on main and double check things work as planned. 

In general I feel it's good to have less things being called in this function and in the rendering that are async (and less things in the state) and possibly racey so nice work with the changes in this pr.